Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -274,69 +300,145 @@ def TestConfigUpdate(Contents):
             retval = -1
     return retval
 
-def SetConfigUpdate(Contents):
-    destFileFullPath = CONFIG_PATH.__add__(DEST_FILE_NAME)
+def TestMultiHoming(ConfigID, Ensure):
+    wsRunningNPM = GetWorkspaceRunningNPM()
+    if wsRunningNPM is None:
+        if os.path.exists(PATH_NPM_STATE):
+            LG().Log(LogType.Info, 'NPM State directory exists when NPM is not running, set to clean up')
+            return -1
+        if Ensure == 'Present':
+            LG().Log(LogType.Info, 'NPM should be started for workspace ' + ConfigID + ' set to set it up')
+            return -1
+    elif not os.path.exists(PATH_NPM_STATE):
+        LG().Log(LogType.Info, 'NPM State directory deleted when NPM is running, set to check setup')
+        return -1
+    if wsRunningNPM == ConfigID and Ensure == 'Absent':
+        LG().Log(LogType.Info, 'NPM should be purged, set to purge')
+        return -1
+    return 0
+
+def SetMultiHoming(ConfigID, Ensure):
+    wsRunningNPM = GetWorkspaceRunningNPM()
+    retval = 0
+
+    if wsRunningNPM is None:
+        if os.path.exists(PATH_NPM_STATE):
+            LG().Log(LogType.Info, 'NPM State directory exists when NPM is not running, now removed')
+            shutil.rmtree(PATH_NPM_STATE, ignore_errors=True)
```

Comment:
@dineshkarora Added ignore_errors=True here to make sure we do not get errors as remove failures will not affect us. If an error does occur then it will be taken care in SetupNPM properly.